code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm

```diff
@@ -578,11 +578,336 @@
 	duration = 10
 
 #undef MOOK_ATTACK_NEUTRAL
 #undef MOOK_ATTACK_WARMUP
 #undef MOOK_ATTACK_ACTIVE
 #undef MOOK_ATTACK_RECOVERY
 #undef ATTACK_INTERMISSION_TIME
+
+////Jungle Seedling////
+
+#define SEEDLING_STATE_NEUTRAL 0
+#define SEEDLING_STATE_WARMUP 1
+#define SEEDLING_STATE_ACTIVE 2
+#define SEEDLING_STATE_RECOVERY 3
+
+#define SEEDLING_STATE_REGENERATING 4
+#define SEEDLING_REGENERATION_TIME 30
+
+
+/mob/living/simple_animal/hostile/jungle/seedling
+	name = "seedling"
+	desc = "This oversized, predatory flower conceals what can only be described as an organic energy cannon, and it will not die until its hidden vital organs are sliced out. \
+	 The concentrated streams of energy it sometimes produces require its full attention, attacking it during this time will prevent it from finishing its attack."
+	maxHealth = 100
+	health = 100
+	melee_damage_lower = 30
+	melee_damage_upper = 30
+	icon = 'icons/mob/jungle/arachnid.dmi'
+	icon_state = "seedling"
+	icon_living = "seedling"
+	icon_dead = "seedling_dead"
+	pixel_x = -16
+	pixel_y = -14
+	minimum_distance = 3
+	move_to_delay = 20
+	aggro_vision_range = 15
+	ranged = TRUE
+	ranged_cooldown_time = 10
+	projectiletype = /obj/item/projectile/seedling
+	projectilesound = 'sound/weapons/pierce.ogg'
+	robust_searching = TRUE
+	stat_attack = UNCONSCIOUS
+	anchored = TRUE
+	var/combatant_state = SEEDLING_STATE_NEUTRAL
+	var/obj/seedling_weakpoint/weak_point
+	var/datum/beam/solar_beam = null
+	var/mob/living/beam_debuff_target
+	var/solar_beam_identifier = 0
+
+/obj/item/projectile/seedling
+	name = "solar energy"
+	icon_state = "seedling"
+	damage = 10
+	damage_type = BURN
+	light_range = 2
+	flag = "energy"
+	light_color = LIGHT_COLOR_YELLOW
+	hitsound = 'sound/weapons/sear.ogg'
+	hitsound_wall = 'sound/weapons/effects/searwall.ogg'
+	nondirectional_sprite = TRUE
+
+/obj/item/projectile/seedling/Bump(atom/A, yes)//Stops seedlings from destroying other jungle mobs through FF
+	if(isliving(A))
+		var/mob/living/L = A
+		if("jungle" in L.faction)
+			return FALSE
+	return ..()
+
+/obj/effect/ebeam/solarbeam
+	name = "beam of solar energy"
+
+/datum/status_effect/seedling_beam_indicator
+	id = "seedling beam indicator"
+	duration = 30
+	status_type = STATUS_EFFECT_MULTIPLE
+	alert_type = null
+	tick_interval = 1
+	var/mutable_appearance/beam_overlay
+	var/atom/target
+	var/angle = 0
+
+/datum/status_effect/seedling_beam_indicator/on_creation(mob/living/new_owner, target_plant)
+	. = ..()
+	if(.)
+		target = target_plant
+		tick()
+
+/datum/status_effect/seedling_beam_indicator/on_apply()
+	beam_overlay = mutable_appearance('icons/mob/jungle/arachnid.dmi', "seedling_beam_indicator")
+	beam_overlay.pixel_x = -16 -owner.pixel_x
+	beam_overlay.pixel_y = -16 -owner.pixel_y
+	tick()
+	return ..()
+
+/datum/status_effect/seedling_beam_indicator/Destroy()
+	if(owner)
+		owner.overlays -= beam_overlay
+	QDEL_NULL(beam_overlay)
+	return ..()
+
+/datum/status_effect/seedling_beam_indicator/tick()
```

Comment:
also the indicator looks ugly anyway is this really necessary?
If it is, remove it for now, when 512 comes along there's a way to do this in a cheaper manner, and you can have it then.